src/main/java/org/sagebionetworks/web/client/widget/FullWidthAlert.java

```diff
@@ -0,0 +1,145 @@
+
+public class FullWidthAlert implements IsWidget {
+	public interface Binder extends UiBinder<Alert, FullWidthAlert> {
+	}
+
+	private static Binder uiBinder = GWT.create(Binder.class);
+
+	@UiField
+	Icon icon;
+	@UiField
+	Span messageSpan;
+	@UiField
+	Strong messageTitle;
+	@UiField
+	Button primaryButton;
+	@UiField
+	Button secondaryButton;
+	
+	Alert w;
+
+	/**
+	 * This is a full width info Alert component, with an icon, message and link.
+	 * 
+	 * ## Usage In your ui.xml, how to add the InfoAlert.
+	 * 
+	 * You can customize the icon, the message and the link target. ```
+	 * xmlns:w="urn:import:org.sagebionetworks.web.client.widget"
+	 * <w:InfoAlert icon="CHECK_CIRCLE" message="x files were added to your Download List" linkTarget=""
+	 * linkText="view download list" /> ```
+	 */
+	public FullWidthAlert() {
+		w = uiBinder.createAndBindUi(this);
+	}
+
+	@Override
+	public Widget asWidget() {
+		return w;
+	}
+
+	public void setVisible(boolean visible) {
+		w.setVisible(visible);
+	}
+
+	public void setAddStyleNames(String styleNames) {
+		w.addStyleName(styleNames);
+	}
+
+	public boolean isVisible() {
+		return w.isVisible();
+	}
+
+	public boolean isAttached() {
+		return w.isAttached();
+	}
+
+	public void setIcon(IconType type) {
+		icon.setType(type);
+	}
+
+	public void setMessageTitle(String title) {
+		messageTitle.setVisible(true);
+		messageTitle.setText(title);
+	}
+	public void setMessage(String message) {
+		messageSpan.setText(message);
+	}
+
+	public void addPrimaryCTAClickHandler(ClickHandler c) {		
+		primaryButton.addClickHandler(c);
+	}
+
+	public void setPrimaryCTAText(String text) {
+		primaryButton.setVisible(true);
+		primaryButton.setText(text.toUpperCase());
+	}
+	public void setPrimaryCTAButtonType(ButtonType type) {
+		primaryButton.setType(type);
+	}
+
+	public void setPrimaryCTAHref(String href) {
+		primaryButton.setHref(href);
+		addPrimaryCTAClickHandler(event -> {
+			Window.open(href, "_blank", "");
+		});
+	}
+
+	public void addSecondaryCTAClickHandler(ClickHandler c) {		
+		secondaryButton.addClickHandler(c);
+	}
+
+	public void setSecondaryCTAText(String text) {
+		secondaryButton.setVisible(true);
+		secondaryButton.setText(text.toUpperCase());
+	}
+	public void setSecondaryCTAButtonType(ButtonType type) {
+		secondaryButton.setType(type);
+	}
+
+	public void setSecondaryCTAHref(String href) {
+		secondaryButton.setHref(href);
+		addSecondaryCTAClickHandler(event -> {
+			Window.open(href, "_blank", "");
+		});
+
+	}
+
+	public void setAlertType(AlertType type) {
+		IconType defaultIcon;		
+		switch(type) {
+			case DANGER: defaultIcon = IconType.EXCLAMATION_CIRCLE;
+				break;
+			case INFO: defaultIcon = IconType.INFO;
+				break;
+			case WARNING: defaultIcon = IconType.BELL;
+				break;
+			case SUCCESS: 
+			case DEFAULT: 
+			default: defaultIcon = IconType.CHECK;
+		}
+		icon.setType(defaultIcon);
+		w.setType(type);
+	}
+
+	public void setDismissable(boolean dismissable) {
```

Comment:
I'm guessing we typically use whatever the default value for `dismissable` is for the Alert component? Should we lift the default up to this level for visibility?

Reply:
Not sure what you mean, could you please elaborate on your recommendation?

Reply:
It's not clear to me what the default value for `dismissable` is. Unless I missed something, it's not specified here or in the corresponding ui.xml. So my assumption is that gwtbootstrap3's Alert specifies the default value.

Just wondering if we should set a default value somewhere in our code (particularly if design has a preference). Ultimately not a big deal.